factory-containers/build.sh

```diff
@@ -100,23 +101,32 @@ for x in $IMAGES ; do
 		# sanity check and pull in a cached image if it exists. if it can't be pulled set no_op_tag to 0.
 		run docker pull ${ct_base}:latest || no_op_tag=0
 		if [ $no_op_tag -eq 0 ] && [ -z "$CHANGED" ]; then
-			echo "WARNING - no cached image found, forcing a rebuild"
+			status "WARNING - no cached image found, forcing a rebuild"
 		fi
 		auth=1
 	fi
 
-	cd $x
 	if [ $no_op_tag -eq 1 ] ; then
 		status Tagging docker image $x for $ARCH
 		run docker tag ${ct_base}:latest ${ct_base}:$TAG-$ARCH
 	else
+		docker_cmd="docker build --label \"jobserv_build=$H_BUILD\" -t ${ct_base}:$TAG-$ARCH --force-rm"
 		if [ -z "$NOCACHE" ] ; then
 			status Building docker image $x for $ARCH with cache
-			run docker build --label "jobserv_build=$H_BUILD" --cache-from ${ct_base}:latest -t ${ct_base}:$TAG-$ARCH --force-rm .
+			docker_cmd="$docker_cmd  --cache-from ${ct_base}:latest"
 		else
 			status Building docker image $x for $ARCH with no cache
-			run docker build --label "jobserv_build=$H_BUILD" --no-cache -t ${ct_base}:$TAG-$ARCH --force-rm .
+			docker_cmd="$docker_cmd  --no-cache"
 		fi
+
+		DOCKERFILE="${DOCKERFILE-Dockerfile}"
+		if [ -n "$BUILD_CONTEXT" ] ; then
+			status "Using custom build context $BUILD_CONTEXT"
+			BUILD_CONTEXT="$REPO_ROOT/$x/${BUILD_CONTEXT}"
```

Comment:
hrm -- build context is relative to the $x/ directory, as opposed to the $REPO_ROOT?

In my current example, I'd need to set `BUILD_CONTEXT=../` -- is that an intended constraint?

Reply:
honestly - i was confused which directory it should be relative to. My thinking was to make it relative to the directory the docker-build.conf file lived in. But yes, I think `BUILD_CONTEXT=../` shoud "just work".